taskcluster/scriptworker_taskgraph/transforms/python_version.py

```diff
@@ -0,0 +1,60 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Create a task per job python-version
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+from copy import deepcopy
+
+from taskgraph.transforms.base import TransformSequence
+
+
+transforms = TransformSequence()
+
+def _replace_string(obj, repl_dict):
+    if isinstance(obj, dict):
+        for k in obj.keys():
+            obj[k] = obj[k].format(**repl_dict)
+    elif isinstance(obj, list):
+        for c in range(0, len(obj)):
+            obj[c] = obj[c].format(**repl_dict)
+    else:
+        obj = obj.format(**repl_dict)
+    return obj
+
+
+@transforms.add
+def tasks_per_python_version(config, jobs):
+    for job in jobs:
+        for python_version in job.pop("python-versions"):
+            task = deepcopy(job)
+            repl_dict = {"name": job["name"], "python_version": python_version}
+            task["label"] = _replace_string(task["label"], repl_dict)
+            task['worker']['docker-image'] = _replace_string(task['worker']['docker-image'], repl_dict)
+            task['description'] = _replace_string(task['description'], repl_dict)
+            if task.get('run', {}).get('command'):
+                task['run']['command'] = _replace_string(task['run']['command'], repl_dict)
+            if task['worker'].get('command'):
+                task['worker']['command'] = _replace_string(task['worker']['command'], repl_dict)
+            if task.get('docker-repo'):
+                task['docker-repo'] = _replace_string(task['docker-repo'], repl_dict)
```

Comment:
I think this would be a bit clearer with a loop like [here](https://searchfox.org/mozilla-central/rev/027893497316897b8f292bde48dbb6da2391a331/taskcluster/taskgraph/transforms/l10n.py#233-258) over the fields being modified.